src/extension.ts

```diff
@@ -247,3 +249,42 @@ function registerCommands(context: ExtensionContext) {
     });
     context.subscriptions.push(restartServer);
 }
+
+function configureLanguage(context: ExtensionContext) {
+    const disposable = languages.setLanguageConfiguration('rust', {
+        onEnterRules: [
+            {
+                // Begins a triple-slash doc comment
+                // e.g. ///| or //!|
+                beforeText: /^\s*\/{2}(\/|\!).*$/,
+                action: { indentAction: IndentAction.None, appendText: '/// ' },
```

Comment:
Currently this always inserts `/// `, even if in a module comment `//! `.
Would it be possible to specify separate rules for `///` and `//!` respectively? Didn't toy with that myself, so I have no idea if that's easy/possible at all.

Reply:
You're right. Fixed. Thanks for the catch